primitives/src/validator.rs

```diff
@@ -67,6 +85,12 @@ impl TryFrom<&str> for ValidatorId {
             value
         };
 
+        if hex_value.len() != 40 {
```

Comment:
please use match with the other case above, instead of splitting it on multiple checks/lines.
Also I think that more explicit errors will help even us when debugging errors.

Reply:
`match` wouldn't be cleaner in this scenario, and joining the cases together would make it further confusing

Reply:
although I have changed the error message to make it more descriptive